baseband_tasks/sampling.py

```diff
@@ -402,11 +407,12 @@ class SampleShift(PaddedTaskBase):
     Resample : resample a stream to a new grid, without time shifts
     """
     def __init__(self, ih, shift, *, samples_per_frame=None):
-        shift = self.shift = np.array(shift).astype(int, casting='safe')
+        shift = self._shift = np.round(to_sample(ih, shift)).astype(int)
         check_broadcast_to(shift, ih.sample_shape)
+        start_time = ih.start_time + shift.max() / ih.sample_rate
         super().__init__(ih, pad_start=0, pad_end=shift.ptp(),
-                         samples_per_frame=samples_per_frame)
-        self._start_time += shift.max() / ih.sample_rate
+                         samples_per_frame=samples_per_frame,
+                         start_time=start_time)
```

Comment:
Just thinking about the best way to bookkeep the original time after the shifts. Should we provide a property to recover the original time?

Reply:
Right now, this is the way it is done everywhere, and one would have to do `fh.ih.start_time` to get the previous time. This maybe something we can end up using metadata for. But not in this PR!